juju/juju
# Allow machine agents to connect to debuglog endpoint

Previously only users could request the debug log, but we want to use it as the source for the migration log transfer, so the machine agent needs to be able to request it as well. 

Added `httpContext.stateForRequestAuthenticatedTag`, which allows the calling code to say what kinds of entities should be allowed. Also generalised `apiserver.common.AuthEither` into `AuthAny`, which can combine any number of auth functions.

Moved the base debuglog tests into `debuglog_db_test.go` to make them easier to find - the file version of the debuglog is gone now.

QA steps:
Bootstrapped and checked that I could still use debug-log. It's hard to check that a machine agent can use it, but I'll be checking that in my logtrasfer testing.
